internal/httputil/http.go

```diff
@@ -0,0 +1,76 @@
+package httputil // import "github.com/pomerium/pomerium/internal/httputil"
+
+import (
+	"context"
+	"fmt"
+	stdlog "log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/pomerium/pomerium/internal/log"
+	"github.com/pomerium/pomerium/internal/urlutil"
+)
+
+// NewHTTPServer starts a http server given a set of options and a handler.
+//
+// It is the caller's responsibility to Close() or Shutdown() the returned
+// server.
+func NewHTTPServer(opt *ServerOptions, h http.Handler) *http.Server {
```

Comment:
Standardized interface (and defaults) for any HTTP server we want to ListenAndServe from given a set of handlers and options. That way we don't have a bunch of different servers with different timeouts, loggers, etc running around.